File: db4oj.tests/src/com/db4o/db4ounit/marshall/YapStringTestCase.java
package com.db4o.db4ounit.marshall;

import com.db4o.*;

import db4ounit.*;
import db4ounit.extensions.*;

public class YapStringTestCase extends AbstractDb4oTestCase {

	public void testIndexMarshalling() {
		YapReader reader=new YapReader(2*YapConst.INT_LENGTH);
		YapStream stream=(YapStream)db();
		YapString handler=new YapString(stream,stream.stringIO());
		final int[] original = new int[]{0xdb,0x40};
		handler.writeIndexEntry(reader,original);
		reader._offset=0;
		int[] retrieved=(int[])handler.readIndexEntry(reader);
		ArrayAssert.areEqual(original,retrieved);
	}
	
}

File: db4on/test/src/com/db4o/db4ounit/common/handlers/YapStringTestCase.cs
namespace com.db4o.db4ounit.common.handlers
{
	public class YapStringTestCase : Db4oUnit.Extensions.AbstractDb4oTestCase
	{
		public virtual void TestIndexMarshalling()
		{
			com.db4o.YapReader reader = new com.db4o.YapReader(2 * com.db4o.YapConst.INT_LENGTH
				);
			com.db4o.YapStream stream = (com.db4o.YapStream)Db();
			com.db4o.YapString handler = new com.db4o.YapString(stream, stream.StringIO());
			com.db4o.inside.slots.Slot original = new com.db4o.inside.slots.Slot(unchecked((int
				)(0xdb)), unchecked((int)(0x40)));
			handler.WriteIndexEntry(reader, original);
			reader._offset = 0;
			com.db4o.inside.slots.Slot retrieved = (com.db4o.inside.slots.Slot)handler.ReadIndexEntry
				(reader);
			Db4oUnit.Assert.AreEqual(original._address, retrieved._address);
			Db4oUnit.Assert.AreEqual(original._length, retrieved._length);
		}
	}
}
